pkg/apis/istio/v1alpha1/istioinstaller_types.proto

```diff
@@ -0,0 +1,585 @@
+syntax = "proto3";
+
+package v1alpha1;
+
+import "k8s.io/api/core/v1/generated.proto";
+import "k8s.io/api/autoscaling/v1/generated.proto";
+import "k8s.io/apimachinery/pkg/apis/meta/v1/generated.proto";
+import "github.com/gogo/protobuf/protobuf/google/protobuf/wrappers.proto";
+
+/*
+IstioInstall is a schema for both defining and overlaying Istio control plane installations.
+The Istio operator uses a compiled-in InstallSpec as default installation values, along with compiled-in install charts.
+Running the operator with an empty user defined InstallSpec results in an installation with default values, using the
+default charts.
+
+The simplest install specialization is to point the user InstallSpec baseValuesPath to a different values file, for
+example an Istio auth installation, which will use the values associated with the auth installation profile for Istio.
+
+Further specialization are possible at three levels:
+
+1. New APIs defined in this file
+
+Feature API: this API groups an Istio install by features and allows enabling/disabling the features, as well as some
+additional high level settings that are feature specific. Each feature contains one or more components, which
+correspond to Istio components in the cluster.
+
+k8s API: this API is a pass through to k8s resource settings for Istio k8s resources. It allows customizing Istio k8s
+resources like Affinity, Resource requests/limits, PodDisruptionBudgetSpec, Selectors etc. in a more consistent and
+k8s specific way compared to values.yaml. See KubernetesResourcesSpec in this file for details.
+
+2. values.yaml
+
+The entirety of values.yaml settings is accessible through InstallSpec. They are included for backward compatibility,
+but also because many of the values will be moved to configuration CRs that are consumed directly by Istio components
+rather than being install time parameters. The only remaining fields will be k8s resource settings which will be
```

Comment:
This last sentence is confusing.... I think it is confusing to make "will" statement in the future.  It is not clear why the k8s resource settings will be remaining fields in values.yaml if they are moved to k8s resources API.